burr/core/application.py

```diff
@@ -1957,6 +1971,24 @@ def _repr_mimebundle_(self, include=None, exclude=None, **kwargs):
         dot = self.visualize(include_conditions=True, include_state=False)
         return dot._repr_mimebundle_(include=include, exclude=exclude, **kwargs)
 
+    def validate_correct_async_use(self):
+        """Validates that the application is meant to run async."""
+        # This is a gentle warning for existing users
+        if self._adapter_set.async_hooks:
+            logger.warning(
+                "There are asynchronous hooks present in the application that will be ignored. "
+                "Please use async methods to run the application and have them executed. "
+                f"The application has following asynchronous hooks: {self._adapter_set.async_hooks} "
+            )
+
+        # Seems fair to raise this if everything is async but the app execution is sync
+        if self._adapter_set.async_hooks and self._builder.is_async:
```

Comment:
Confused -- this is erroring out if the builder is async and the adapter has async hooks? `.is_async`?

Reply:
Yes! So I am calling this in the sync methods and it  covers cases 3 and 7 in the above table. Thinking back, maybe it's better to be strict from the get go and only check `_builder.is_async` and error out.

Previously, if there were no async hooks present (i.e. no async persisters, etc.) it doesn't really matter how the app was built or is run -- since only sync hooks in the app. But might as well just say if you build it async, use it async.